### DISCUSSION TITLE
Anyone willing to mentor a rust port?

### DISCUSSION THREAD

**Discussion by DrAlta** [markqvist/Reticulum#585] [Help & Questions]
I'm developing a Disruption Tolerant Network, and it looks like Reticulum does everything I want. but I'm not sure I can make a compatible implementation in Rust by myself.

I've just started to have the time available to work on this and have barely started to read the documentation so don't know how much filling in the gaps I'd need

Then there is the issue that both implementations have to be kept up to date with any changed to the protocol.

**Comment by teschmitt** [comment]
There are already two Rust implementations:

1. https://github.com/BeechatNetworkSystemsLtd/Reticulum-rs - I suspect this one is geared towards the requirements of Beechat but after a first glance at the source it looks somewhat general.
2. https://lib.rs/crates/reticulum - alpha but worth a try.

With Reticulum at v1.0.0 I wouldn't expect breaking changes anytime soon. I'd fork one of the above implementations suited best for my use-case and go from there.

**Comment by cobraPA** [comment]
3?  My understanding is Beechat has one as well.

**Reply by teschmitt** [reply]
BeeChat's implementation is the first one I linked.

**Comment by worst-git-server** [comment]
I'm currently working on an implementation in elixir, so I could help a bit.
